src/nuclei/client/main.py

```diff
@@ -183,11 +183,18 @@ def _get_app_specification(self, app: str, version: str = "latest") -> dict:
             self.get_url(app, version) + "/openapi.json", timeout=self.timeout
         )
         if response.status_code != 200:
-            raise ConnectionError(
-                "Unfortunately the application you are trying to reach is unavailable (status code: "
-                f"{response.status_code}). Please check you connection. If the problem persists contact "
-                "CEMS at info@cemsbv.nl"
+            error_message = (
+                f"Failed to get the API specification for {app} version {version}.\n"
+                + f"Status code: {response.status_code}. Response: {response.text}"
             )
+            if response.status_code < 500:
```

Comment:
There's more status codes than 200 which are acceptable, and everything in the 300 range shouldn't be regarded as an error either.